src/components/ui/Panel.jsx

```diff
@@ -91,41 +96,41 @@ export default class Panel extends React.Component {
    */
   stopResize = event => {
     event.preventDefault();
-    clearTimeout(this.timer);
 
-    const nativeEvent = event.nativeEvent;
-    const clientY = nativeEvent.changedTouches
-      ? nativeEvent.changedTouches[0].clientY
-      : nativeEvent.clientY;
-    const sizes = [0, window.innerHeight / 2, window.innerHeight];
-    const positionIndex = getClosestIndex(sizes, clientY);
+    document.removeEventListener('mousemove', this.moveCallback);
+    document.removeEventListener('touchmove', this.moveCallback);
+
+    const { previousSize, currentHeight } = this.state;
 
     let size = null;
-    if (positionIndex === 2 && this.holding) {
-      size = 'minimized';
-    } else if (positionIndex === 1 && !this.holding) {
-      size = 'minimized';
-    } else if (positionIndex === 0) {
-      size = 'maximized';
+    if (event.timeStamp - this.interactionStarted < CLICK_RESIZE_TIMEOUT_MS) {
+      // the resize handler was only clicked, provide 'smart' resize
+      size = !previousSize ? 'minimized' : null;
+    } else {
+      // the resize handler was really dragged-n-dropped
+      const sizes = [0, window.innerHeight / 2, window.innerHeight];
+      const positionIndex = getClosestIndex(sizes, currentHeight);
+      if (positionIndex === 0) {
```

Comment:
What about positionIndex === 1 ?

Reply:
It's the medium position, i.e. the default case, which would result in `size=null`.

Reply:
ok, shouln't we asign `size = null` in this case ?

Reply:
It's already the initial value, line 105.